orion-lib/src/listeners/filterchain.rs

```diff
@@ -170,19 +170,32 @@ impl FilterchainType {
         stream: AsyncStream,
         downstream_metadata: Arc<DownstreamConnectionMetadata>,
         shard_id: ThreadId,
-        listener_name: &'static str,
+        listener_name: &str,
```

Comment:
Why has this changed ? Because of this change you need to clone listener_str everywhere ..